enterprise/bloom-index/src/main/java/org/neo4j/kernel/api/impl/bloom/BloomIndexTransactionEventUpdater.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.impl.bloom;
+
+import org.apache.lucene.document.Document;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.neo4j.graphdb.Entity;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.NotFoundException;
+import org.neo4j.graphdb.event.PropertyEntry;
+import org.neo4j.graphdb.event.TransactionData;
+import org.neo4j.graphdb.event.TransactionEventHandler;
+
+public class BloomIndexTransactionEventUpdater implements TransactionEventHandler<Object>
+{
+    private final WritableDatabaseBloomIndex nodeIndex;
+    private final WritableDatabaseBloomIndex relationshipIndex;
+    private final String[] properties;
+
+    BloomIndexTransactionEventUpdater( WritableDatabaseBloomIndex nodeIndex, WritableDatabaseBloomIndex relationshipIndex, String[] properties )
+    {
+        this.nodeIndex = nodeIndex;
+        this.relationshipIndex = relationshipIndex;
+        this.properties = properties;
+    }
+
+    @Override
+    public Object beforeCommit( TransactionData data ) throws Exception
```

Comment:
Looks like we don't separate removed properties from updated properties. Do we have tests that verify that remove properties work?